src/skin/qml/qmlskin.cpp

```diff
@@ -104,12 +119,34 @@ QWidget* QmlSkin::loadSkin(QWidget* pParent,
         mixxx::CoreServices* pCoreServices) const {
     Q_UNUSED(pConfig);
     Q_UNUSED(pSkinCreatedControls);
-    Q_UNUSED(pCoreServices);
     VERIFY_OR_DEBUG_ASSERT(isValid()) {
         return nullptr;
     }
 
     qmlRegisterType<QmlControlProxy>("Mixxx", 0, 1, "ControlProxy");
+    qmlRegisterType<QmlWaveformOverview>("Mixxx", 0, 1, "WaveformOverview");
+
+    qmlRegisterSingletonType<QmlPlayerManagerProxy>("Mixxx",
+            0,
+            1,
+            "PlayerManager",
+            lambda_to_singleton_type_factory_ptr(
+                    [pCoreServices](QQmlEngine* pEngine,
+                            QJSEngine* pScriptEngine) -> QObject* {
+                        Q_UNUSED(pScriptEngine);
+
+                        QmlPlayerManagerProxy* pPlayerManagerProxy =
+                                new QmlPlayerManagerProxy(
```

Comment:
Passing the plain pointer that is actually managed by shared_ptr<> and not by Qt is dangerous! Then we should instead consistently use std::shared_ptr<PlayerManager>.

Reply:
No, instead we should tell the Qml Engine that it's managed by the JS engine.